### DISCUSSION TITLE
request help: 关于替换微服务方案(spring cloud gateway + nacos)的请教

### DISCUSSION THREAD

**Discussion by whiteketdev** [apache/apisix#5195] [General]
### Issue description

1.目前公司考虑apisix替代目前微服务方案(spring cloud gateway + nacos)，这块是否可行，有成功的案例吗？
2.如果还想替换nacos有什么好的推荐吗？
3.高可用我看是通过etcd集群。如果是对apisix做高可用呢有什么方法吗？

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

**Comment by tzssangglass** [comment]
> 1.目前公司考虑apisix替代目前微服务方案(spring cloud gateway + nacos)，这块是否可行，有成功的案例吗？

can be replaced.

> 2.如果还想替换nacos有什么好的推荐吗？

This is related to your actual needs and the cost of replacement. You can learn about consul.

> 3.高可用我看是通过etcd集群。如果是对apisix做高可用呢有什么方法吗？

APISIX is inherently stateless and high availability of APISIX is high availability of ETCD.
